packages/x-components/src/x-installer/x-installer/x-installer.ts

```diff
@@ -292,6 +294,25 @@ export class XInstaller {
     }
   }
 
+  protected normaliseSnippetConfig(snippetConfig: SnippetConfig): NormalisedSnippetConfig;
+  protected normaliseSnippetConfig(snippetConfig: Partial<SnippetConfig>): Partial<SnippetConfig>;
+  /**
+   * Transforms the snippet configuration.
+   * - If `lang` is provided and `uiLang` is not, it sets `uiLang=lang`.
+   *
+   * @param snippetConfig - The snippet config to normalise.
+   * @returns The normalised version of the given snippet config.
+   * @internal
+   */
+  protected normaliseSnippetConfig(
+    snippetConfig: SnippetConfig | Partial<SnippetConfig>
+  ): NormalisedSnippetConfig | Partial<SnippetConfig> {
+    if (snippetConfig.lang) {
+      snippetConfig.uiLang ??= snippetConfig.lang;
+    }
```

Comment:
The watcher approach, while ideal to have in one place the side effects of mutating the snippet config brought some issues, as you lost the information of which properties where modified, so instead I opted to do a more imperative approach in the only method that had all the information we needed to make a decision.